packages/versioning/src/versioning.ts

```diff
@@ -171,6 +237,38 @@ export function getNameAtVersion(p: Program, t: Type, v: ObjectType): string {
   return "";
 }
 
+/**
+ * @returns get old type if applicable.
+ */
+export function getTypeBeforeVersion(p: Program, t: Type, v: ObjectType): any {
+  const target = toVersion(p, t, v);
+  const map = getTypeChangedFrom(p, t);
+  if (!map || !target) return "";
+
+  for (const [key, val] of map) {
+    if (target.index < key.index) {
+      return val;
+    }
+  }
+  return "";
```

Comment:
```suggestion
  return undefined;
```

I think those 2 make more sense here

Reply:
Fixed